### PR TITLE
Makefile install fixes

### PR DESCRIPTION
**Pull request by laenion**
This pull request tries to fix two Makefile bugs of the `install` target:
1. On 32 Bit x86 machines the ignition binaries were searched in the wrong directory due to a missing Go architectures <-> Linux hardware name; another mapping was introduced, handling various machine strings such as _i486_ or _i686_.
2. Dracut and systemd are all installed with the default permissions of _755_. Separate the files by extension and assign the matching permissions so the executable bit is only set for scripts. This will also avoid warning by systemd due to the incorrect permissions.

### CODE REVIEW

**Comment by jlebon** [comment]
> On 32 Bit x86 machines the ignition binaries were searched in the wrong directory due to a missing Go architectures <-> Linux hardware name; another mapping was introduced, handling various machine strings such as i486 or i686.

BTW, are you actually planning to use Ignition on 32-bit platforms? For FCOS/RHCOS at least, we don't currently support that and don't plan to. It's possible you'll hit other snags along the way.

File: Makefile [hunk 1]
```diff
@@ -14,11 +14,17 @@ all:
...
+	  bn=$$(basename $$bn); \
+	  install -m 0644 -D -t $(DESTDIR)/usr/lib/dracut/modules.d/$${bn} $$x; \
 	done
-	install -D -t $(DESTDIR)/usr/lib/systemd/system systemd/*
+	for x in dracut/*/*.sh dracut/*/*-generator; do \
+	  bn=$$(dirname $$x); \
+	  bn=$$(basename $$bn); \
+	  install -m 0755 -D -t $(DESTDIR)/usr/lib/dracut/modules.d/$${bn} $$x; \
+	done
+	install -m 0644 -D -t $(DESTDIR)/usr/lib/systemd/system systemd/*
```
[the remark this answers is not in the document]
Comment (laenion) [reply Makefile:24]: Sure, definitely looks better - I've modified the patch.

**Comment by laenion** [comment]
> BTW, are you actually planning to use Ignition on 32-bit platforms? For FCOS/RHCOS at least, we don't currently support that and don't plan to. It's possible you'll hit other snags along the way.

Not really, as we are not building MicroOS or Kubic for any 32 bit platform, either. However - as they are sharing their repository with Tumbleweed which supports i586 - I didn't see a reason to explicitly prevent it from building on x86. We don't have any automated tests of the x86 build, but Ignition seemed to be working fine so far, and I will see build failures (it's the second one, the first one was #944)...

**Comment by laenion** [comment]
We also got a community patch for 32 bit ARM machines.

**Pull request merged**